lib/Doctrine/Migrations/Configuration/Configuration.php

```diff
@@ -140,6 +143,24 @@ public function getMigrationsColumnLength() : int
         return $this->migrationsColumnLength;
     }
 
+    public function setMigrationsExecutedAtColumnName(string $migrationsExecutedAtColumnName) : void
+    {
+        $this->migrationsExecutedAtColumnName = $migrationsExecutedAtColumnName;
+    }
+
+    public function getMigrationsExecutedAtColumnName() : string
+    {
+        return $this->migrationsExecutedAtColumnName;
+    }
+
+    public function getQuotedMigrationsExecutedAtColumnName() : string
+    {
+        return $this->getDependencyFactory()
```

Comment:
To be honest, this thing is just a service locator now... Wiring everything should imho be done in the SF Bundle / Zend Module and/or in a factory/builder object.

Reply:
It can be done, but not without a big BC break. The only place it is being used like a service locator is in places where I wanted to maintain BC. Like here. Also, these are all internal services so it feels weird to put the burden on the end user to wire all these classes up that they should not even know about or have access to.

Reply:
To be clear, I am referring to developers who don't use Symfony, Zend, etc. and have their app integrated with Doctrine DBAL and Migrations. Today all that is required to configure migrations is setting up one service that is passed your DBAL connection:

```
$configuration = new Configuration($connection);
// configure your migrations
$configuration->...;
```

In 2.0 this is the same. All the existing setups will just continue to work. All the services that make up the internals are not meant to be public facing services. If I put the wiring in the framework integrations, wouldn't this mean that when I refactor the internals, it'd be a BC break in the framework integrations due to the wiring would needing to be updated?